statsd/container_linux.go

```diff
@@ -198,13 +198,16 @@ func readCIDOrInode(userProvidedID, cgroupPath, selfMountInfoPath, defaultCgroup
 	}
 
 	if cgroupFallback {
-		if isHostCgroupNs {
-			containerID = readContainerID(cgroupPath)
+		containerID = readContainerID(cgroupPath)
+		if containerID != "" {
 			return
 		}
+
 		containerID = readMountinfo(selfMountInfoPath)
-		if containerID == "" {
-			containerID = getCgroupInode(defaultCgroupMountPath, cgroupPath)
+		if containerID != "" || isHostCgroupNs {
```

Comment:
What's the purpose of checking `isHostCgroupNs` here?

Reply:
I think we don't want to send the cgroup inode if we're in host cgroup namespace. Otherwise we'll send it even if the app is not in a container

Reply:
I see, there isn't a reliable way to detect that. I think the approximation saying if `/proc/self/cgroup` does not find anything while being **in host cgroup namespace**, you can return.
There should be a dedicated code path and comment for it.

Reply:
Shouldn't an app that is running directly on host run in the host cgroup ns ? I am not sure I understood where exactly we should move this check but I added a dedicated `if` condition with a comment right after reading mountinfo.

Reply:
Yes that's correct. You can move the check before `mountinfo` but it does not really matter.